src/components/Main/Results/ResultsCard.tsx

```diff
@@ -26,19 +26,19 @@ interface ResultsCardProps {
   autorunSimulation: boolean
   toggleAutorun: () => void
   canRun: boolean
-  params: AllParams
   mitigation: ContainmentData
   severity: SeverityTableRow[] // TODO: pass severity throughout the algorithm and as a part of `AlgorithmResult` instead?
   result?: AlgorithmResult
   caseCounts?: EmpiricalData
   scenarioUrl?: string
 }
 
+const MemoizedDeterministicLinePlot = React.memo(DeterministicLinePlot)
+
```

Comment:
This reduces the UI lag considerably. I couldn't really figure out why DeterministicLinePlot is so slow to render; but memoizing it seems to solve the problem nicely.